### ISSUE TITLE
Changes to `_config.yml` not included in rebuild on jekyll-3.0.0.pre.beta8

### ISSUE THREAD

**Issue by Simbul** [jekyll/jekyll#3854]
I was playing around with the current `master` (currently 56622c7ab617741c765521e2aacfc59bfd8f7319) and I think I found a regression.

To reproduce, create a dummy site and then:

1. Build the site with `jekyll build`
2. Change a parameter (e.g. `title`) in `_config.yml`
3. Rebuild the site: the new value for `title` is not picked up

I had a look at the insides of the gem and, while I can't say I have a perfect understanding of how it works, it looks like changes in `_config.yml` do not cause pages to be marked as "to be regenerated".

I built the gem following the [Pre-releases instructions](http://jekyllrb.com/docs/installation/#pre-releases). 
Regeneration works fine with the current stable build (jekyll-2.5.3).

/cc @sparrovv who I was pairing with when I encountered the issue

**Comment by envygeeks** [comment]
The configuration is not re-read upon change, see these comment for context:
https://github.com/jekyll/jekyll/issues/2302#issuecomment-42323080

I'm going to leave this open though because even though I agree with @parkr's reasoning behind why he chose to do this I think we need to tell the user something along the lines: "Your configuration has changed, restart to pick up the changes" or something along those lines because there can be a lot of confusion for new users.

**Comment by Simbul** [comment]
I think making that clearer would be great and I see why you wouldn't `watch` for changes in configuration.

This issue is slightly different, however: changes in `_config.yml` were not picked up even when killing the server, running `jekyll build` and restarting the server. Moreover, that behaviour works fine in 2.5.3.

**Comment by envygeeks** [comment]
Can you provide a snapshot of the source or a repo that I can clone to test this behavior?

**Comment by envygeeks** [comment]
Wait, I think I know what's going on.  Delete the file `.jekyll-metadata` and see if it updates, if it does then it's related to the regenerator, which I do not believe triggers on a change in `_config.yml` (I could be wrong on that and @alfredxing would have to confirm.)

**Comment by Simbul** [comment]
Yes. If I delete `.jekyll-metadata`, the subsequent `jekyll build` will pick up the config changes.
However, it will also generate a new metadata file, which will cause config changes do be ignored again from that point on.

**Comment by envygeeks** [comment]
The latter part is a redundant sentiment given the success of the test.  The behavior is unintuitive but now that we know where it's at I think we can and should fix it. /cc @jekyll/core

**Comment by Simbul** [comment]
Here are detailed instructions to reproduce.

First of all, [compile the gem from source](http://jekyllrb.com/docs/installation/#pre-releases):

```
$ git clone git://github.com/jekyll/jekyll.git
$ cd jekyll
$ script/bootstrap
$ bundle exec rake build
$ ls pkg/*.gem | head -n 1 | xargs gem install -l
```

Then build an new empty project:

```
$ jekyll new jekyll-regression
...
$ cd jekyll-regression/
$ jekyll build
...
$ head -9 _site/index.html
<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Your awesome title</title>
```

"Your awesome title" is the default value in `_config.yml`.

Here's the bug, where nothing changes when something in the config file changes:

```
$ sed -i -e 's/Your awesome title/Changed title/g' _config.yml
$ jekyll build
...
$ head -9 _site/index.html
<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Your awesome title</title>
```

Update is unblocked by removing the metadata file:

```
$ rm .jekyll-metadata
$ jekyll build
...
$ head -9 _site/index.html
<!DOCTYPE html>
<html>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Changed title</title>
```

**Comment by envygeeks** [comment]
This is ***not*** a regression, this is a bug in the regenerator which is *new* and *beta* in order for this to be a regression we would have to of stopped loading your config, of which we did not do, in this instance there is a bug where the config does not trigger a regeneration.

**Comment by Simbul** [comment]
I apologise for my lack of knowledge of the project history. I just assumed it was a regression because it is working fine in 2.1.3, but I'm just as happy to call it a bug now that I know that's more accurate :smiley:

**Comment by seantcanavan** [comment]
This is still an issue for me. 

jekyll 3.6.0
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
ubuntu 16.04

I have to manually delete `.jekyll-metadata` after making changes to `_config.yml` in order to see those changes. I'm executing with:

`alias jsi='jekyll serve --incremental'`

**Comment by pathawks** [comment]
@seantcanavan This issue is unrelated. Please open a new issue.
